src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -432,16 +435,41 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       placement.inlineSize,
     ]);
 
+    /**
+     * Returns true if the AppLayout is nested
+     * Does not apply to iframe
+     */
+    const getIsNestedInAppLayout = (element: HTMLElement | null): boolean => {
+      let currentElement: Element | null = element?.parentElement ?? null;
+
+      while (currentElement) {
+        if (getComputedStyle(currentElement).getPropertyValue(globalVars.stickyVerticalTopOffset)) {
+          return true;
+        }
+        currentElement = currentElement.parentElement;
```

Comment:
CSS var values propagate. You do not need to manually walk up

Reply:
Technically, the decision suggested by you is better, but it does not work in u tests env so I turned it back

Reply:
Okay, so `getComputedStyle` implementation is JSDOM is more naive.

Let's leave a comment for the future readers this loop is needed for JSDOM. 

Fine to keep it because in the real browsers it will exit after the first iteration anyway